webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,311 @@
+import React from 'react';
+import './ticketStyle.scss';
+
+import {Dispatch} from 'redux';
+
+import {Instance} from 'types/model';
+
+export type Props = {
+    href: string;
+    show: boolean;
+    connected: boolean;
+    isLoaded?: boolean;
+    ticketDetails?: TicketDetails;
+    defaultUserInstanceID?: string;
+    connectedInstances: Instance[];
+    getIssueByKey: (issueKey: string, instanceID: string) => (dispatch: Dispatch, getState: any) => Promise<{
+        data?: TicketData;
+    }>;
+    setTicket?: (ticketDetails: {}) => void;
+}
+
+export type State = {
+    href: string;
+    isLoaded: boolean;
+    ticketId: string;
+    ticketDetails?: TicketDetails;
+};
+
+const isAssignedLabel = ' is assigned';
+const unAssignedLabel = 'Unassigned';
+const jiraTicketTitleMaxLength = 80;
+
+enum myStatus {
+    INDETERMINATE = 'indeterminate',
+    DONE = 'done',
+}
+
+const myStatusClasses: Record<string, string> = {
+    [myStatus.INDETERMINATE]: ' ticket-status--indeterminate',
+    [myStatus.DONE]: ' ticket-status--done',
+};
+
+export default class TicketPopover extends React.PureComponent<Props, State> {
+    truncateString(str: string, num: number) {
+        if (num > str.length) {
+            return str;
+        }
+        return `${str.substring(0, num)}...`;
+    }
+
+    constructor(Props: Props) {
+        super(Props);
+        const ticketID = this.getIssueKey();
+
+        this.state = {
+            href: this.props.href,
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    getIssueKey = () => {
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
```

Comment:
I think we should catch errors here, since there are many indexes that can go wrong and it will fail if one of those is not there.